posthog/models/hog_functions/hog_function.py

```diff
@@ -0,0 +1,115 @@
+import json
+from typing import Optional
+
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch.dispatcher import receiver
+
+from posthog.models.action.action import Action
+from posthog.models.team.team import Team
+from posthog.models.utils import UUIDModel
+from posthog.redis import get_client
+
+
+class HogFunction(UUIDModel):
+    team: models.ForeignKey = models.ForeignKey("Team", on_delete=models.CASCADE)
+    name: models.CharField = models.CharField(max_length=400, null=True, blank=True)
+    description: models.TextField = models.TextField(blank=True, default="")
+    created_at: models.DateTimeField = models.DateTimeField(auto_now_add=True, blank=True)
+    created_by: models.ForeignKey = models.ForeignKey("User", on_delete=models.SET_NULL, null=True, blank=True)
+    deleted: models.BooleanField = models.BooleanField(default=False)
+    updated_at: models.DateTimeField = models.DateTimeField(auto_now=True)
+    enabled: models.BooleanField = models.BooleanField(default=False)
+
+    hog: models.TextField = models.TextField()
+    bytecode: models.JSONField = models.JSONField(null=True, blank=True)
+
+    # TODO: Rename to "variables"
+    inputs_schema: models.JSONField = models.JSONField(null=True)
+    inputs: models.JSONField = models.JSONField(null=True)
+    filters: models.JSONField = models.JSONField(null=True, blank=True, default=dict)
+
+    @property
+    def filter_action_ids(self) -> list[int]:
+        try:
+            return [int(action["id"]) for action in self.filters.get("actions", [])]
+        except KeyError:
+            return []
+
+    def compile_filters_bytecode(self, actions: Optional[dict[int, Action]] = None):
+        from .utils import hog_function_filters_to_expr
+        from posthog.hogql.bytecode import create_bytecode
+
+        if actions is None:
+            # If not provided as an optimization we fetch all actions
+            actions_list = (
+                Action.objects.select_related("team").filter(team_id=self.team_id).filter(id__in=self.filter_action_ids)
+            )
+            actions = {action.id: action for action in actions_list}
+
+        try:
+            self.filters["bytecode"] = create_bytecode(hog_function_filters_to_expr(self.filters, self.team, actions))
+        except Exception:
+            # TODO: Better reporting of this issue
+            self.filters["bytecode"] = None
```

Comment:
```suggestion
            # TODO: Better reporting of this issue
            self.filters["bytecode"] = None
            self.filters["bytecode_error"] = str(e.msg)
```

Something simple like this to debug in production if filters hit the fan... even if it's not exposed anywhere?

Reply:
Yeah I'll add it in